File: solutions/go/2017/01/main.go
package main

/*
* ElfScript Brigade
*
* Advent Of Code 2017 Day 01
* Rust Solution
*
* Day 1: Inverse Captcha
*
* https://adventofcode.com/2017/day/1
*
 */

import (
	"github.com/luxedo/esb_fireplace-go"
	"strconv"
	"strings"
)

func solve(captcha string, steps int) int {
	captcha2 := captcha[steps:] + captcha[0:steps]
	acc := 0
	for idx, curr := range captcha {
		next := captcha2[idx]
		if rune(next) == curr {
			num, _ := strconv.Atoi(string(next))
			acc += num
		}
	}
	return acc
}

func solve_pt1(input_data string, args []string) (interface{}, error) {
	captcha := strings.Trim(input_data, "\n")
	return solve(captcha, 1), nil
}

func solve_pt2(input_data string, args []string) (interface{}, error) {
	captcha := strings.Trim(input_data, "\n")
	return solve(captcha, len(captcha)/2), nil
}

func main() {
	// 🎅🎄❄️☃️🎁🦌
	// Bright christmas lights HERE
	esb_fireplace.V1Run(solve_pt1, solve_pt2)
}

File: solutions/haskell/2017/01/app/Main.hs
{-  ElfScript Brigade
 -
 -  Advent Of Code 2017 Day 01
 -  Haskell Solution
 -
 -  Day 1: Inverse Captcha
 -
 -  https://adventofcode.com/2017/day/1 -}
module Main where

import           Data.Char (digitToInt, isSpace)
import           Fireplace

solvePt1 :: String -> [String] -> Int
solvePt1 input _args = do
  let captcha = filter (not . isSpace) input
  sum $ map (digitToInt . fst) $ filter (uncurry (==)) $ zip captcha (drop 1 $ cycle captcha)

solvePt2 :: String -> [String] -> String
solvePt2 _input _args = do
  "December"

main :: IO ()
main = do
  -- 🎅🎄❄️☃️🎁🦌
  -- Bright christmas lights HERE
  v1Run solvePt1 solvePt2
